src/main/java/com/commercetools/sync/commons/utils/RetryableSphereClientWithExponentialBackoff.java

```diff
@@ -0,0 +1,215 @@
+        this.retryErrorStatusCodes = DEFAULT_RETRY_ERROR_STATUS_CODES;
+    }
+
+    /**
+     * Creates a new instance of {@link RetryableSphereClientWithExponentialBackoff} given a {@link SphereClientConfig}
+     * responsible for creation of a SphereClient.
+     *
+     * @param clientConfig the client configuration for the client.
+     * @return the instantiated {@link RetryableSphereClientWithExponentialBackoff}.
+     */
+    public static RetryableSphereClientWithExponentialBackoff of(
+            @Nonnull final SphereClientConfig clientConfig) {
+        return new RetryableSphereClientWithExponentialBackoff(clientConfig);
+    }
+
+    /**
+     * Sets the timeout value.
+     * @param timeout - build with timeout value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeout value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeout(final long timeout) {
+        this.timeout = timeout;
+        return this;
+    }
+
+    /**
+     * Sets the initialDelay value.
+     * @param initialDelay - build with initialDelay value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given initialDelay value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withInitialDelay(final long initialDelay) {
+        if (initialDelay < timeout) {
+            this.initialDelay = initialDelay;
+        }
+        return this;
+    }
+
+    /**
+     * Sets the timeUnit.
+     * @param timeUnit - build with timeUnit value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given timeUnit value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withTimeUnit(@Nonnull final TimeUnit timeUnit) {
+        this.timeUnit = timeUnit;
+        return this;
+    }
+
+    /**
+     * Sets the Max Retry value.
+     * @param maxRetries - build with maxRetries value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxRetries value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxRetries(final int maxRetries) {
+        if (maxRetries > 1) {
+            this.maxRetries = maxRetries;
+        }
+        return this;
+    }
+
+    /**
+     * Sets the Max Parallel Requests value.
+     * @param maxParallelRequests - build with maxParallelRequests value.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given maxParallelRequests value.
+     */
+    public RetryableSphereClientWithExponentialBackoff withMaxParallelRequests(final int maxParallelRequests) {
+        this.maxParallelRequests = maxParallelRequests;
+        return this;
+    }
+
+    /**
+     * Sets the Retry Error Status Codes.
+     * @param retryErrorStatusCodes - build with retryErrorStatusCodes.
+     * @return {@link RetryableSphereClientWithExponentialBackoff} with given retryErrorStatusCodes.
+     */
+    public RetryableSphereClientWithExponentialBackoff withRetryErrorStatusCodes(
```

Comment:
not used in anywhere.

Reply:
Added in tests. https://github.com/commercetools/commercetools-sync-java/pull/623/commits/75550c89a7c5affbb9f32af5cf72bb52756c8ec4